osf/models/user.py

```diff
@@ -1662,6 +1661,10 @@ def add_unclaimed_record(self, claim_origin, referrer, given_name, email=None):
             'email': clean_email,
         }
         self.unclaimed_records[pid] = record
+
+        self.save()  # must save for PK to add system tags
+        self.add_system_tag(unregistered_created_source_tag(referrer_id))
```

Comment:
Ah, I see why you remove the `.save()` after `.add_unclaimed_record()` is called. Ignore my requests since it is relevant. And I assume the formatting comes from the linter.